week1/homework/src/server.js

```diff
@@ -2,14 +2,47 @@
 
 const http = require('http');
 
-/* `createServer` MUST return an instance of `http.Server` otherwise the tests
- * will fail.
+/* `createServer` MUST return an instance of `http.Server` 
+otherwise the tests will fail.
  */
-function createServer(port) {
+function createServer() {
   let state = 10;
 
   const server = http.createServer((request, response) => {
     // TODO: Write your homework code here
+    switch (request.url) {
+      case '/state':
+        const stateJson1 = { 'state': state };
+        response.writeHead(200, { 'Content-Type': 'application/json' });
+        response.end(JSON.stringify(stateJson1));
```

Comment:
Why create a `const` variable every time for the response?  

You could also use 

`response. write(JSON.stringify({ 'state': state }));`

We usually use `response.write` to write to the client and use `response.end` to indicate the end.